### PR TITLE
[3/N] Fixes clang-tidy warnings in header files

### PR DESCRIPTION
**Pull request by cyyever**
This PR series tries to enable clang-tidy for headers in torch/csrc and c10/util.

### CODE REVIEW

File: torch/csrc/utils/object_ptr.h [hunk 1]
```diff
@@ -8,9 +8,7 @@ class TORCH_PYTHON_API THPPointer {
  public:
   THPPointer() : ptr(nullptr){};
   explicit THPPointer(T* ptr) noexcept : ptr(ptr){};
-  THPPointer(THPPointer&& p) noexcept {
-    free();
-    ptr = p.ptr;
+  THPPointer(THPPointer&& p) noexcept : ptr(p.ptr) {
     p.ptr = nullptr;
```
Comment (Skylion007) [inline torch/csrc/utils/object_ptr.h:12]: Is it safe here to no longer do the free?

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Fix memleak

File: torch/csrc/utils/object_ptr.h (hunk 1 above)
Comment (cyyever) [reply torch/csrc/utils/object_ptr.h:12]: What confused me is the free() in a constructor, in this case the pointer is nullptr.

**Comment by cyyever** [comment]
@Skylion007 The move constructor initialized the pointer to nullptr. Nothing to free.

File: torch/csrc/utils/object_ptr.h [hunk 2]
```diff
@@ -8,9 +8,7 @@ class TORCH_PYTHON_API THPPointer {
...
   };
```
Comment (Skylion007) [inline torch/csrc/utils/object_ptr.h:13]:
```suggestion
  THPPointer(THPPointer&& p) noexcept : ptr(std::exchange(p.ptr, nullptr)) {}
```
is a bit more cleaner and does what you want I think.

**Review by Skylion007 — APPROVED** [review]
Looks good, but I would use std::exchange to make the move ctor a bit cleaner.

**Pull request closed without merging**